guides/plugins/plugins/content/cms/add-cms-element-via-admin-sdk.md

```diff
@@ -0,0 +1,352 @@
+
+{% hint style="info" %}
+This example uses TypeScript, which is recommended, but not required for developing Shopware.
+{% endhint %}
+
+{% page-ref page="../../administration/app-base-guide.md" %}
+{% page-ref page="../../administration/add-custom-component.md" %}
+
+## Creating your custom element
+
+Like in the Plugin Guide for this topic, imagine you want to create a new element to display a Dailymotion video. The shop manager can configure the link of the video to be shown. That's exactly what you're going to build in this guide.
+
+Creating a new element via app requires you to use the Admin Extension SDK.
+
+### Target structure
+
+It is yours to decide which approach to use when creating apps, since everything we're doing here will be loaded via iFrame, but since Shopware's best practice is a full Vue.js approach, this tutorial will do exactly that.
+
+When our extension is finished, you'll get the following file structure:
+
+{% code title="<plugin root>/src/Resources/app/administration/src" %}
+
+```bash
+├── base
+│   └── mainCommands.ts
+├── main.ts
+├── viewRenderer.ts
+└── views
+    └── swag-dailymotion
+        ├── swag-dailymotion-config.ts
+        ├── swag-dailymotion-element.ts
+        └── swag-dailymotion-preview.ts
+```
+
+{% endcode %}
+
+## Initial loading of components
+
+Everything starts in the `main.ts` file:
+
+```ts
+import 'regenerator-runtime/runtime';
+import { location } from '@shopware-ag/admin-extension-sdk';
+
+// Only execute extensionSDK commands when
+// it is inside a iFrame (only needed for plugins)
+if (location.isIframe()) {
+    if (location.is(location.MAIN_HIDDEN)) {
+        // Execute the base commands
+        import('./base/mainCommands');
+    } else {
+        // Render different views
+        import('./viewRenderer');
+    }
+}
+```
+
+This is main file, which is executed first and functions as our entry point.
+
+We start with `if(location.isIframe())` to make sure we only load content used inside iFrames. The reason for this is that the SDK can be used in apps and plugins, so this check makes sure that your code is executed in the right place.
+
+Next we need `if(location.is(location.MAIN_HIDDEN))` to **load the main commands**, which are defined in the `mainCommands.ts` file. This will only be used to load logic, but not templates into the Administration.
+
+Lastly the `else` case will be responsible for specific loading of views via `viewRenderer.ts`. This is where our view templates will be loaded.
+
+### Loading all required templates
+
+Now we need to create the `viewRenderer.ts` file, which will include the three mandatory files needed for a CMS element. We prefix them with `swag-dailymotion`, consisting of a vendor prefix and the component name, to make sure no other developer will accidentally choose the same name as we did:
```

Comment:
Replace the info as below :

Now, create the `viewRenderer.ts` file, which will include the three mandatory files needed for a CMS element. Prefix them with `swag-dailymotion`, consisting of a vendor prefix and the component name to make sure no other developer will accidentally choose the same name: